### ISSUE TITLE
Implement histogram resampling

### ISSUE THREAD

**Issue by maxnoe** [cta-observatory/pyirf#49]
For many calculations we will need resampling the finely binned IRFs to coarser bins for actual computation.

We should implement (a) function(s) to do this.

A possible approach for this is interpolating the cumulative sum.

**Comment by adonath** [comment]
We also use this method in Gammapy, but it's hard-coded in our IRF classes where we need it. However it's not much code to re-implement. Here is an example implementation that I did some time ago to illustrate the method: https://github.com/adonath/notebooks-public/blob/master/histogram_resampling.ipynb 

Using `scipy.interpolate.interp1d` or possibly `RegularGridInterpolator `this can be extended to handle ND data as well, depending on how many axis you would like to integrate. In general I think this method is the best compromise between performance and precision. Using a suitable scaling for the data e.g. log-log for energy axes or sqrt for the theta ** 2 axes the accuracy of the method can be improved further...

**Comment by nbiederbeck** [comment]
I would like to tackle this. Is it "enough" to implement the function from the notebook Axel posted? 

Also could you assign me, I am not able to do this myself?

**Comment by LukasNickel** [comment]
I have already done this for the migration matrix without much optimization.
https://github.com/cta-observatory/pyirf/pull/64

If we have a more general method, that code can be used there aswell.

**Comment by HealthyPear** [comment]
Was this issue fixed after #76 ? or do we want to add more implementations?

**Comment by maxnoe** [comment]
I think we can close this for now, not clear if more is needed.
